Amazon.QLDB.Driver/result/AsyncBufferedResult.cs

```diff
@@ -0,0 +1,72 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.QLDB.Driver
+{
+    using System.Collections;
+    using System.Collections.Generic;
+    using Amazon.IonDotnet.Tree;
+
+    /// <summary>
+    /// Implementation of a result which buffers all values in memory, rather than stream them from QLDB during retrieval.
+    /// This implementation should only be used when the result is to be returned after the parent transaction is to be
+    /// committed.
+    /// </summary>
+    public class AsyncBufferedResult : BaseBufferedResult, IAsyncResult
+    {
+        private readonly List<IIonValue> values;
+        private readonly IOUsage? consumedIOs;
+        private readonly TimingInformation? timingInformation;
+
+        /// <summary>
+        /// Prevents a default instance of the <see cref="AsyncBufferedResult"/> class from being created.
+        /// </summary>
+        ///
+        /// <param name="values">Buffer values.</param>
+        /// <param name="consumedIOs">IOUsage statistics.</param>
+        /// <param name="timingInformation">TimingInformation statistics.</param>
+        private AsyncBufferedResult(List<IIonValue> values, IOUsage? consumedIOs, TimingInformation? timingInformation)
```

Comment:
Please remove the implementation here. We still don't know what it will be like just yet.

Reply:
Done.